controllers/managedocs_controller_test.go

```diff
@@ -852,7 +861,41 @@ var _ = Describe("ManagedOCS controller", func() {
 				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
 			})
 		})
-		When("there is a value for PAGERDUTY_KEY in the pagerduty secret and a value for SNITCH_URL in the deadmanssnitch secret", func() {
+		When("there is a notification email address in the add-on parameter secret and no smtp secret", func() {
+			It("should not create alertmanager config", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secretKey := utils.GetResourceKey(secret)
+				Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+
+				// Add notification email to the addon param secret
+				secret.Data["notification-email-0"] = []byte("test@email.com")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				// Create pagerduty secret with valid key
+				pdSecret := pdSecretTemplate.DeepCopy()
+				pdSecret.Data["PAGERDUTY_KEY"] = []byte("test-key")
+
+				// Create deadman's snitch secret with a valid key
+				dmsSecret := dmsSecretTemplate.DeepCopy()
+				dmsSecret.Data["SNITCH_URL"] = []byte("test-key")
+
+				Expect(k8sClient.Create(ctx, pdSecret)).Should(Succeed())
```

Comment:
move each create after the code that set it up, in this case move this to line 877